Source/OpenTK/Platform/Egl/EglContext.cs

```diff
@@ -97,9 +95,7 @@ abstract class EglContext : EmbeddedGraphicsContext
                 window.CreateWindowSurface(config);
 
             int[] attrib_list = new int[] { Egl.CONTEXT_CLIENT_VERSION, major, Egl.NONE };
-            HandleAsEGLContext = Egl.CreateContext(window.Display, config, shared != null ? shared.HandleAsEGLContext : IntPtr.Zero, attrib_list);
-
-            MakeCurrent(window);
+            HandleAsEGLContext = Egl.CreateContext(window.Display, config, sharedContext != null ? (sharedContext as IGraphicsContextInternal).Context.Handle : IntPtr.Zero, attrib_list);
```

Comment:
Sure, I will improve it.
So far, I simply copy-pasted the existing code done for shared context in most other implementations (Windows, Mac, etc...); should I fix those too?